File: apps/backend/teams-service/src/main/java/com/blockout/teams/models/entities/Team.java
package com.blockout.teams.models.entities;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import jakarta.persistence.*;
import java.time.LocalDateTime;

import com.blockout.teams.models.enums.Format;
import com.blockout.teams.models.enums.Gender;

@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "teams", uniqueConstraints = {
        @UniqueConstraint(columnNames = { "club_id", "division_id", "format", "gender", "raw_name",
                "season" }, name = "uix_team")
})
public class Team {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "club_id", nullable = false)
    private String clubId;

    @Column(name = "raw_name", nullable = false)
    private String rawName;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "short_name", nullable = false)
    private String shortName;

    @Column(name = "league_code", nullable = false)
    private String leagueCode;

    @Column(name = "division_id", nullable = false)
    private Long divisionId;

    @Column(name = "season", nullable = false)
    private String season;

    @Enumerated(EnumType.STRING)
    @Column(name = "format", nullable = false)
    private Format format;

    @Enumerated(EnumType.STRING)
    @Column(name = "gender", nullable = false)
    private Gender gender;

    @Builder.Default
    @Column(name = "followers_count", nullable = false)
    private Long followersCount = 0L;

    @Column(name = "logo_url")
    private String logoUrl;

    @Builder.Default
    @Column(name = "active", nullable = false)
    private Boolean active = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "last_update")
    private LocalDateTime lastUpdate;

    @PrePersist
    public void prePersist() {
        createdAt = LocalDateTime.now();
        lastUpdate = LocalDateTime.now();
    }

    @PreUpdate
    public void preUpdate() {
        lastUpdate = LocalDateTime.now();
    }
}

File: apps/backend/competition-scraper/scraper/infrastructure/blockout/team.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class CreateTeamInternalRequest:
    """Exact request accepted by teams-service's create route."""

    clubId: str
    rawName: str
    name: str
    shortName: str
    leagueCode: str
    divisionId: int
    season: str
    format: str | None
    gender: str | None
    followersCount: int | None
    logoUrl: str | None
    active: bool


@dataclass(frozen=True)
class UpdateTeamInternalRequest:
    """Exact request accepted by teams-service's update route."""

    clubId: str
    rawName: str
    name: str
    shortName: str
    leagueCode: str
    divisionId: int
    logoUrl: str | None
    season: str
    format: str | None
    gender: str | None
    active: bool


@dataclass
class TeamInternalResponse:
    """Complete teams-service representation used by scraper decisions."""

    clubId: str
    rawName: str
    name: str
    shortName: str
    leagueCode: str
    divisionId: int
    season: str
    gender: str | None = None
    format: str | None = None
    id: int | None = None
    followersCount: int | None = 0
    logoUrl: str | None = None
    active: bool = True
    createdAt: datetime | None = None
    lastUpdate: datetime | None = None
